src/audio_sdl_mixer.cpp

```diff
@@ -44,6 +44,58 @@
 using namespace std::chrono_literals;
 
 namespace {
+	struct RefHolder {
+		Filesystem::InputStream stream;
+	};
+
+	Sint64 SDLCALL vio_size(struct SDL_RWops * context) {
+		auto stream = reinterpret_cast<RefHolder*>(context->hidden.unknown.data1)->stream;
+		return stream->get_size();
+	}
+
+	Sint64 SDLCALL vio_seek(struct SDL_RWops * context, Sint64 offset, int whence) {
+		auto stream = reinterpret_cast<RefHolder*>(context->hidden.unknown.data1)->stream;
+
+		stream->seekg(offset, Filesystem::CSeekdirToCppSeekdir(whence));
+
+		return stream->tellg();
+	}
+
+	size_t SDLCALL vio_read(struct SDL_RWops * context, void *ptr, size_t size, size_t maxnum) {
+		auto stream = reinterpret_cast<RefHolder*>(context->hidden.unknown.data1)->stream;
+
+		if (size == 0) return 0;
+
+		return stream->read(reinterpret_cast<char*>(ptr), size * maxnum).gcount() / size;
+	}
+
+	size_t SDLCALL vio_write(struct SDL_RWops*, const void*, size_t, size_t) {
+		// Not needed
+		return 0;
+	}
+
+	int SDLCALL vio_close(struct SDL_RWops * context) {
+		// If this is the last shared pointer, the stream get's closed now
+		auto stream_ref = reinterpret_cast<RefHolder*>(context->hidden.unknown.data1);
+		delete stream_ref;
+
+		context->hidden.unknown.data1 = nullptr;
+		return 0;
+	}
+
+	SDL_RWops* create_StreamRWOps(Filesystem::InputStream stream){
+		SDL_RWops * ret = SDL_AllocRW();
+		// create a new shared pointer to avoid deletion of the content when the scope of this function ends
+		ret->hidden.unknown.data1 = new RefHolder { std::move(stream) };
```

Comment:
If this was `unique_ptr`, you could drop an indirection by releasing the `unique_ptr` here and deleting the stream itself in `vio_close`. No need for `RefHolder`